test/run_test.py

```diff
@@ -405,6 +429,9 @@ def run_test(
 
     command = (launcher_cmd or []) + executable + argv
     print_to_stderr("Executing {} ... [{}]".format(command, datetime.now()))
+    if log_file is not None:
```

Comment:
Nit: let's just check `if log_file:` to avoid people passing empty string as log file.  Also, should we check for some exception with open here, i.e. `except (FileNotFoundError, PermissionError, OSError)` so that the shell command can continue if the log file couldn't be open for write?

Reply:
decided to just move the file creation into this function, would that still require checking for exceptions?

Reply:
Well, it would be fine like this IMO